### DISCUSSION TITLE
When doing composer install --no-dev getting error GeneratorsServiceProvider not found

### DISCUSSION THREAD

**Discussion by woutvdpol** [Laravel-Backpack/community-forum#1000] [Q&A (Help)] [answered]
Hi I am making preparations to push my application towards production but when I do the command `composer install --no-dev` I am getting the error:

`
In Application.php line 855:
                                                                   
  Class "Backpack\Generators\GeneratorsServiceProvider" not found  
                                                                   

Script @php artisan storage:link handling the post-install-cmd event returned with error code 1

`

The backpack/generators package is a require-dev package and I don't think I have used it explicitly when creating my application so I was wondering if it is getting used in one of the backpack packages or something?

Thanks

**Comment by pxpm** [comment]
Hello @woutvdpol 

Try searching your app for `GeneratorsServiceProvider`  string. 

It's possible that you added it in the providers list in `config/app.php` or something similar ? 

Let me know, 
Cheers

**Comment by jcastroa87** [comment]
Hello @woutvdpol you were able to solve this case with @pxpm  answer?

Thanks.

**Comment by woutvdpol** [comment]
I don't use the GeneratorsServiceProvider in any of my own classes I believe, I will double check first thing in the morning tomorrow

**Comment by woutvdpol** [comment]
I've just checked it again, and with a fresh install the only mentions of a GeneratorsServiceProvider is in the composer.lock file, so I don't think I just it in any of my own code.
<img width="645" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/25347211/91bfa19f-1f18-4a50-8895-bb955d934b3a">

**Reply by pxpm** [reply]
Hey @woutvdpol let's try to get to the bottom of the issue. 

I've just tried it now, and it worked as expected:
![image](https://github.com/Laravel-Backpack/community-forum/assets/7188159/2c01ee57-b734-416f-9462-35ea77f94971)

It removed all my dev dependencies, and everything worked the same. 

Generators is just an "helper package", none of our other packages use it, only CRUD in dev mode. 

for the sake of testing it from scratch, I deleted my vendor folder and ran `composer install --no-dev`, once again, everything was smooth.

I will need some more information from you to be able to help you. 

Can you share with me your composer.json ? At least the `require` and `require-dev` parts ? 

By any chance are you trying to run `backpack:install` in production? (you shouldn't do that, but it's possible 🤷 )

Cheers

**Reply by woutvdpol** [reply]
I've just removed my vendors map, and ran the command again still the same issue. 
I've copied the composer here
```json
"require": {
        "php": "^8.2",
        "ashallendesign/laravel-exchange-rates": "^7.0",
        "backpack/crud": "^6.0",
        "backpack/permissionmanager": "^7.0",
        "backpack/pro": "^2.0",
        "backpack/theme-coreuiv2": "^1.1",
        "brick/money": "^0.8.1",
        "coderflex/laravel-turnstile": "^2.0",
        "guzzlehttp/guzzle": "^7.7",
        "laravel/framework": "^10.10",
        "laravel/sanctum": "^3.3",
        "laravel/tinker": "^2.8",
        "livewire/livewire": "^3.1",
        "masmerise/livewire-toaster": "^2.0",
        "mews/purifier": "*",
        "mollie/laravel-mollie": "^2.23",
        "monarobase/country-list": "^3.5",
        "spatie/laravel-livewire-wizard": "^2.0.1",
        "spatie/laravel-permission": "^5.10",
        "spatie/laravel-sitemap": "^7.2",
        "srmklive/paypal": "~3.0",
        "symfony/intl": "^7.0"
    },
    "require-dev": {
        "backpack/generators": "^4.0",
        "barryvdh/laravel-debugbar": "^3.8",
        "fakerphp/faker": "^1.9.1",
        "jn-jairo/laravel-ngrok": "^3.0",
        "laracasts/generators": "^2.0",
        "laravel/pint": "^1.13",
        "laravel/sail": "^1.18",
        "mockery/mockery": "^1.4.4",
        "nunomaduro/collision": "^7.0",
        "phpunit/phpunit": "^10.1",
        "spatie/laravel-ignition": "^2.0"
    },
```

I've also tried it locally and it also gives me the error there

**Reply by pxpm** [reply]
@woutvdpol can you put the composer code inside a json tag so that it gets formated? you need to put three (3) backticks and say it's json ```json

```json
{
"require": {
    "backpack/crud": "^6.0"
},
}
```

![image](https://github.com/Laravel-Backpack/community-forum/assets/7188159/c0fe2541-f2cc-4cf4-b872-027b22ce2333)

Thanks

**Reply by woutvdpol** [reply]
Done

**Reply by pxpm** [reply]
Thank you @woutvdpol 🙏 

Hum .. so I could "almost" reproduce your issue, but for that I need to have a mention of the provider somewhere when running `composer install --no-dev`. 

I added the `Backpack/Generators/GeneratorsServiceProvider` to `config/app.php` in the "providers" array for example:

![image](https://github.com/Laravel-Backpack/community-forum/assets/7188159/0d2e01ed-3475-4cf8-a14c-359f917d006a)

As you can see the error is similar, but it's in the providers file, not in in App. So for me, it means you are mentioning the provider somewhere else in the app. 

When this error triggers, if you navigate to `storage/logs/` you will find a more descriptive trace of the error, for example for the one above: 
```bash
[2024-06-03 13:13:59] local.ERROR: Class "Backpack\Generators\GeneratorsServiceProvider" not found {"exception":"[object] (Error(code: 0): Class \"Backpack\\Generators\\GeneratorsServiceProvider\" not found at C:\\laragon\\www\\demo2\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\ProviderRepository.php:206)
[stacktrace]
#0 C:\\laragon\\www\\demo2\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\ProviderRepository.php(142): Illuminate\\Foundation\\ProviderRepository->createProvider('Backpack\\\\Genera...')
#1 C:\\laragon\\www\\demo2\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\ProviderRepository.php(61): Illuminate\\Foundation\\ProviderRepository->compileManifest(Array)
#2 C:\\laragon\\www\\demo2\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(765): Illuminate\\Foundation\\ProviderRepository->load(Array)
#3 C:\\laragon\\www\\demo2\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Bootstrap\\RegisterProviders.php(17): Illuminate\\Foundation\\Application->registerConfiguredProviders()
#4 C:\\laragon\\www\\demo2\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(263): Illuminate\\Foundation\\Bootstrap\\RegisterProviders->bootstrap(Object(Illuminate\\Foundation\\Application))
#5 C:\\laragon\\www\\demo2\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Console\\Kernel.php(447): Illuminate\\Foundation\\Application->bootstrapWith(Array)
#6 C:\\laragon\\www\\demo2\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Console\\Kernel.php(199): Illuminate\\Foundation\\Console\\Kernel->bootstrap()
#7 C:\\laragon\\www\\demo2\\artisan(33): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#8 {main}
"} 
```

What's in your logs ?

**Reply by woutvdpol** [reply]
When I search the application the GeneratorsServiceProvider only comes forward in the vendor map.
This is the error log:

```
[2024-06-03 13:51:27] local.ERROR: Class "Backpack\Generators\GeneratorsServiceProvider" not found {"exception":"[object] (Error(code: 0): Class \"Backpack\\Generators\\GeneratorsServiceProvider\" not found at .../vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php:206)
[stacktrace]
#0 .../vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(142): Illuminate\\Foundation\\ProviderRepository->createProvider('Backpack\\\\Genera...')
#1 .../vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(61): Illuminate\\Foundation\\ProviderRepository->compileManifest(Array)
#2 .../vendor/laravel/framework/src/Illuminate/Foundation/Application.php(769): Illuminate\\Foundation\\ProviderRepository->load(Array)
#3 .../vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/RegisterProviders.php(17): Illuminate\\Foundation\\Application->registerConfiguredProviders()
#4 .../vendor/laravel/framework/src/Illuminate/Foundation/Application.php(263): Illuminate\\Foundation\\Bootstrap\\RegisterProviders->bootstrap(Object(Illuminate\\Foundation\\Application))
#5.../vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(447): Illuminate\\Foundation\\Application->bootstrapWith(Array)
#6 .../vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(199): Illuminate\\Foundation\\Console\\Kernel->bootstrap()
#7 .../artisan(35): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#8 {main}
"} 
```

The ... are the same on every row just removed it to not dox myself any further :P

I have also double checked but I don't mention the GeneratorsServiceProvider in the app.php or anywhere else

<img width="639" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/25347211/753c673c-45bf-4cf9-9fce-c00987f459bb">
This is the result when I do a search of the whole applciation - the ones that are in the logs

**Reply by pxpm** [reply, marked as answer]
Hey @woutvdpol 

No problem, always feel free to remove any information you are not comfortable sharing as long the important stays 👍 

I would bet the issue is in that `packages.php`, what file is that ? Where is that file located, what does that file do ? 

Let me know. 

Cheers

**Reply by woutvdpol** [reply]
Ah, I think that was it, it cached the need for the package and I just deleted the packages.php file and did composer install --no-dev and now it works without an error. Weird, thx for your time

**Reply by woutvdpol** [reply]
Still getting the errors on my production site though, but I'll try to check with my local environment if there are any differences

EDIT: Had to remove the vendor map and the packages.php file, then it worked on prod

**Reply by pxpm** [reply]
try running `php artisan optimize:clear` on your production site to try to bust those caches.
